airbyte-integrations/connectors/source-microsoft-teams/source_microsoft_teams/client.py

```diff
@@ -83,6 +84,9 @@ def _get_access_token(self) -> str:
         else:
             raise MsalServiceError(error=result.get("error"), error_description=result.get("error_description"))
 
+    @backoff.on_exception(
+        backoff.expo, (requests.exceptions.ConnectionError, MsalServiceError, requests.exceptions.RequestException), max_tries=7
```

Comment:
could you add a log message which says we are backing off? e.g: `Encountered exception when querying the Microsoft API: <exception>. Backing off`.